lib/puppetserver/ca/utils/config.rb

```diff
@@ -19,6 +19,42 @@ def self.munge_alt_names(names)
           end.sort.uniq.join(", ")
         end
 
+        def self.puppet_confdir
+          if self.running_as_root?
+            '/etc/puppetlabs/puppet'
+          else
+            "#{ENV['HOME']}/.puppetlabs/etc/puppet"
+          end
+        end
+
+        def self.puppetserver_confdir
+          if self.running_as_root?
+            '/etc/puppetlabs/puppetserver'
+          else
+            "#{ENV['HOME']}/.puppetlabs/etc/puppetserver"
+          end
+        end
+
+        def self.default_ssldir
+          File.join(self.puppet_confdir, 'ssl')
+        end
+
+        def self.old_default_cadir
+          File.join(self.puppet_confdir, 'ssl', 'ca')
+        end
+
+        def self.new_default_cadir
+          File.join(self.puppetserver_confdir, 'ca')
+        end
+
+        def self.symlink_to_old_cadir(current_cadir)
+          return if current_cadir != new_default_cadir
+          # This is only run on setup/import, so there should be no files in the
+          # old cadir, so it should be safe to forcibly remove it (which we need
+          # to do in order to create a symlink).
+          Puppetserver::Ca::Utils::FileSystem.forcibly_symlink(new_default_cadir, old_default_cadir)
```

Comment:
Should add the require for this file, to avoid any order-dependent issues.

Reply:
ahh yes, thanks. so many moving parts haha